### ISSUE TITLE
Force SatDump to give up and save LRIT / HRIT images

### ISSUE THREAD

**Issue by alsldlflglhljlk** [SatDump/SatDump#765]
**What would you want to be added to satdump?**

Currently, if my signal from GOES is intermittent or weak, SatDump may receive some portions of several different images broadcasted by GOES and never finish receiving them. Because it never finished receiving them, SatDump will keep those unfinished images in RAM, hoping to finish the image, even though it will never happen.

**Describe the feature you'd like to have**

I would like to be able to select images and be able to choose whether to delete or save them manually.

**Describe alternatives you've considered**

Currently, the only way to force SatDump to save these images is stopping the HRIT pipeline, saving all the images at once, making SatDump "unresponsive". If my idea is not possible, then giving information about the images (such as when decoding NOAA / Meteor / MetOp HRPT) would keep SatDump from being marked as unresponsive, as well as tell the user what is happening.

**Additional context**

![IMG_3179](https://github.com/user-attachments/assets/af61a8db-9e45-4cd8-98bc-90ddab657500)

![IMG_3180](https://github.com/user-attachments/assets/a7d59822-d2f1-43ae-9472-8dda90d5f3cc)

**Comment by JVital2013** [comment]
Hello again!

Something like this is likely possible, but I'm not sure how necessary it is - and it might verge on bad idea.

SatDump does not lose these images; it saves them when the same channel for the next time slot starts. I have thoroughly tested this - I have not lost an image in months, when the last bug I'm aware of with it was fixed.

The reason why I say having a "force save" is a bad idea, is because sometimes segments will legitimately be delayed for a long time. This does not happen under normal operation, but if GOES is operating in single MESO mode, or if there is a higher volume of EMWIN traffic than normal saturating the downlink, segments can be delayed for quite a while before they are finally transmitted. Why give the option to potentially ruin a good capture in these situations?

I'm not sure what the benefit of an "impatient" button would be , which is effectively what this is. I'm not closing this right away though so other devs can chime in - and you can provide counter points, if you'd like.

That being said, we should re-evaluate that hang when stopping the pipeline. That one will be tracked in #669

**Comment by alsldlflglhljlk** [comment]
If `x` packet is missing from an image and sent a long time later, the Earth will have changed and the new packet will be different from the rest of the image. Instead of having an image with some sections old and some new, isn't it better to scrap the old image and generate a new image from the incoming data? Or am I wrong and the old packet is stored and is the same age as the old image (in which case you're sending an old image very late, which is still not ideal)

**Comment by JVital2013** [comment]
No, it does not mix time slots in any case (at least, not anymore - we fixed that bug).

If an incomplete image is received from GOES, SatDump will wait until it is completed, or the next time slot starts. If the next time slot starts and there's a pending incomplete image, it will save the pending image, then start processing the new one.

In the case I talked about where a segment is delayed, it's not a new image. It's the delayed part of the old image. I agree sending an old image late is not ideal, but that's NOAA PDA, not SatDump doing that from time to time. Besides, it is better to get it late, than not at all.

For comparison: goestools completely discards images that miss a segment (but similar to SatDump, it holds them in memory until the next time slot starts). You just can't tell because goestools has no UI.

I will admit, I'm biased against this idea. It adds complication to the code side, when patience or improving your reception is the real fix. Additionally, it introduces the possibility of confusing users into saving incomplete images that will be completed later.

That being said, we let users shoot themselves in the foot in plenty of other cases, so why not here. And I don't want my biases alone to guide the project (but my vote is no). What are your thoughts @Aang23?

**Comment by 67Robbo** [comment]
Great software thankyou. My 2c on this issue.
For Elektro L4 it always stops a transmission slot with a part image.
Which as you rightly say rectifies itself on the next transmission (3 hours later) but the images are not available for viewing in the file system unless I stop the receiver.
Also of note the small image in the bottom right of the screen stop being displayed after a while so I am not 100% sure images are being processed correctly. They do flash up at the end of the image for a fraction of a second.
Do love the nightly build as the latest one fixes the problem of the Deframer losing sync for Elektro and GK-2A here in New Zealand.

**Comment by alsldlflglhljlk** [comment]
Thanks @67Robbo that gave me an idea. How about, instead of having a save and delete button on the images, have SatDump write the images to the disk as they are being received? Obviously, SatDump shouldn't _fully_ process these while they are still in progress, but composite data would be saved in `product.cbor` so SatDump could fully process them later.

**Comment by 67Robbo** [comment]
A save of something would be good, or if nothing has been received for a period then it gives up that would fix the Elektro-L4 issue.
I also note that after a while of auto reception the small preview display stops working, so I am not not too sure if it is still working.

**Comment by ZbychuButItWasTaken** [comment]
I'll side with @JVital2013 on this one.
Waiting for the start of the next time slot before saving a partial image is the simplest and most effective way to ensure no data is lost, especially with the fact some segments can be delayed.
Adding a system that saves the partial image could in my opinion lead to some unreliability and other issues, simply because it'd add quite a bit of complexity to the process. Especially, since the images wouldn't be processed in that time anyway.
